### DISCUSSION TITLE
Understanding Uno's MVUX template

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno.extensions#1573] [Q&A]
_[From Discord](https://discord.com/channels/372137812037730304/550416151172087808/1119181434008109127):_

Ok, the more I read about MVU, the more I don't understand Uno's MVUX template.

To make it clear up front, I want to create my UI's in Xaml, not pure C#. If I wanted to do that, I would stick with Flutter.

What exactly is the difference between MVU and MVUX or is there none?

Can I use Xaml code with MVUX and if so, how? If not, I'll probably just stick with MVVM.

I had hoped that MVUX would be a bit easier to implement than MVVM or would cause a bit less overhead.

Can someone clarify this for me?

**Comment by juanyacovino77** [comment]
First of all, there is no complete documentation, tutorial or sample app for learning MVUX with UNO Platform. So we are fighting ghost here
Second, yes I would like to write UI only with c#, if possible.
Third, is it possible to use MVU frameworks just like Comet or MauiReactor within UNO platform?

**Comment by nickrandolph** [comment]
Hi @juanyacovino77 the MVUX documentation is being reworked as we speak. The [current MVUX documentation](https://platform.uno/docs/articles/external/uno.extensions/doc/Overview/Mvux/Overview.html) is all still accurate but is being redeveloped to provide a better introduction to MVUX. 

The work in progress documentation is available here: https://github.com/unoplatform/uno.extensions/blob/main/doc/Overview/Mvux/Overview.md

Whilst MVUX isn't like Comet/MauiReactor that rebuilds the UI with each iteration, we think that MVUX will be a better balance between leveraging the data binding capabilities of the platform with the reactive style programmming that's made MVU so popular.

**Reply by juanyacovino77** [reply]
Those are greats news!

I read a little bit on the in-progress documentation and it is more easy to follow. I would only suggest to always add a diagram at the top of the doc to show the big picture of the concept you are about to describe
